src/main/scala/com/amazon/deequ/analyzers/MaxLength.scala

```diff
@@ -42,5 +48,19 @@ case class MaxLength(column: String, where: Option[String] = None)
 
   override def filterCondition: Option[String] = where
 
-  private def criterion: Column = length(conditionalSelection(column, where)).cast(DoubleType)
+  private def criterion(nullBehavior: NullBehavior): Column = {
+    nullBehavior match {
+      case NullBehavior.Fail =>
+        val colLengths: Column = length(conditionalSelection(column, where)).cast(DoubleType)
+        conditionalSelection(colLengths, Option(s"${column} IS NULL"), replaceWith = Double.MaxValue)
+      case NullBehavior.Empty =>
+        length(conditionalSelection(col(column), Option(s"${column} IS NULL"), replaceWith = "")).cast(DoubleType)
+      case _ => length(conditionalSelection(column, where)).cast(DoubleType)
+    }
+  }
+  private def getNullBehavior(): NullBehavior = {
```

Comment:
nit: The `()` can be removed. In idiomatic Scala, `()`, called an empty paren(theses) method, denotes that function performs some side effect. Like calling a service or even just logging. Here, we are just extracting data from an object, and therefore, we can remove this `()` from the method.